satpy/tests/test_composites.py

```diff
@@ -170,86 +169,147 @@ def setUp(self):
                            coords={'y': [0, 1], 'x': [0, 1]})
         ds4.attrs['name'] += '4'
         ds4.attrs['resolution'] = 500
-        self.ds4 = ds4
 
         # high resolution version - but too big
-        ds4 = xr.DataArray(da.ones((4, 4), chunks=2, dtype=np.float64),
-                           attrs=attrs.copy(), dims=('y', 'x'),
-                           coords={'y': [0, 1, 2, 3], 'x': [0, 1, 2, 3]})
-        ds4.attrs['name'] += '4'
-        ds4.attrs['resolution'] = 500
-        ds4.attrs['rows_per_scan'] = 1
-        ds4.attrs['area'] = AreaDefinition('test', 'test', 'test',
-                                           {'proj': 'merc'}, 4, 4,
-                                           (-2000, -2000, 2000, 2000))
-        self.ds4_big = ds4
+        ds4_big = xr.DataArray(da.ones((4, 4), chunks=2, dtype=np.float64),
+                               attrs=attrs.copy(), dims=('y', 'x'),
+                               coords={'y': [0, 1, 2, 3], 'x': [0, 1, 2, 3]})
+        ds4_big.attrs['name'] += '4'
+        ds4_big.attrs['resolution'] = 500
+        ds4_big.attrs['rows_per_scan'] = 1
+        ds4_big.attrs['area'] = AreaDefinition('test', 'test', 'test',
+                                               {'proj': 'merc'}, 4, 4,
+                                               (-2000, -2000, 2000, 2000))
+
+        return ds1, ds2, ds3, ds4, ds4_big
 
-    def test_bad_color(self):
-        """Test that only valid band colors can be provided."""
+    @pytest.mark.parametrize(
+        ("case", "exp"),
+        [
+            ("high bad color", ValueError),
+            ("neutral bad color", ValueError),
+            ("match_data_arrays", satpy.composites.IncompatibleAreas),
+            ("more than three datasets", ValueError),
+            ("no high res band in self sharpened", ValueError)
+        ]
+    )
```

Comment:
At this point it might be better to just have separate tests, but...

What if you did this as parameters `("init_kwargs", "call_kwargs", "init_exc", "call_exc")`. But this would require some "fun" context managers to wrap `pytest.raises` with the boolean like `if init_exc is not None:`. Or maybe you can provide an empty tuple to `pytest.raises`...I'm not sure.

Or you could split these into `test_init_errors` and `test_call_errors`. Or split them into separate tests.

If you can't describe the cases as changes in input (even if you use a data producing function as one of the parameters) then they should probably be separate tests.